source/contributing/style-guide.md

```diff
@@ -16,119 +15,104 @@ The same holds for communication directed to contributors and maintainers:
+  Clarify identity if you use "we".
+  Generally, "we" are users and contributors in the Nix ecosystem.
 
-### Headers
-Reserve the largest header (`#`) for the title.
+## Links
 
-Use Markdown headers `##` through `####` to divide up content in the body of the document, but prefer to stay in the `##`-`###` range.
-Finer grained headings are not necessarily better.
-
-### One line per sentence
-Write one sentence per line.
-This makes review easier since the git diffs are line-oriented and the smallest level of granularity in the GitHub review interface is a line of text.
-
-### Links
-
-Unless explicitly required to point to the latest version of an external resource, all references should be [permanent links].
+Unless explicitly required to point to the latest version of an external resource, all references should be [permanent links] to ensure that the referenced content is what is intended.
 
 Many web services offer permalinks, such as:
 
 - [GitHub URLs to specific commits]
 - [Wikipedia URLs to specific page versions]
 - [Internet Archive "Save Page Now" for persisting web pages]
 
-Use [reference links][ref_links] to keep the source legible.
-All links in a section should be grouped together at the end.
-For instance:
+[permanent links]: https://en.wikipedia.org/wiki/Permalink
+[GitHub URLs to specific commits]: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files
+[Wikipedia URLs to specific page versions]: https://en.wikipedia.org/wiki/Wikipedia:Linking_to_Wikipedia#Permanent_links_to_old_versions_of_pages
+[Internet Archive "Save Page Now" for persisting web pages]: https://web.archive.org/save
+
+## Markdown
 
-```
-## This is a section
-Lorem ipsum dolor sit amet, [consectetur][adipiscing] elit. Suspendisse rutrum ligula porta, condimentum dui dignissim, imperdiet mi. Sed interdum lacus nec varius posuere. Duis auctor varius purus, ut ornare purus tempus eu. Aliquam erat volutpat. Etiam eget nunc malesuada, elementum neque eget, mollis metus. Nulla suscipit felis nec accumsan fermentum.
+- Write one sentence per line.
 
-Integer volutpat erat sem, non varius turpis facilisis eu. Nam eu [ullamcorper][magna]. Morbi iaculis vel urna in condimentum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec tellus sit amet tellus venenatis porta in et ex. Nunc sodales nisl magna, at dictum diam sollicitudin id. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec facilisis, sapien eu faucibus iaculis, nibh nibh condimentum enim, nec egestas turpis erat nec libero. Curabitur ut tincidunt odio. Praesent sed tincidunt tortor.
+  This makes long sentences immediately visible, and makes it easier to review changes and make suggestions.
 
-[adipiscing]: example.com
-[magna]: example.com
-```
+  The rule is unambiguous and does not require tooling support to be applied easily.
+  [Here is a discussion of different line wrapping styles.]
```

Comment:
I don't think the `.` should be part of the link, here (if accepted, change line 101 too)

```suggestion
  [Here is a discussion of different line wrapping styles].
```

Reply:
```suggestion
  Here is [a discussion of different line wrapping styles] for reference on the relative merits.
```